draft-ietf-quic-transport.md

```diff
@@ -2326,9 +2390,7 @@ The NEW_CONNECTION_ID is as follows:
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 |                          Sequence (i)                       ...
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                                                               |
-+                        Connection ID (64)                     +
-|                                                               |
+|   Length (8)  |          Connection ID (32..144)            ...
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
```

Comment:
Does it really make sense to support multiple different length Connection IDs for the same connection? Why not just require all Connection IDs be the same length as the original, and then remove this length field altogether? I'm just thinking of the additional complexity of multiple different length Connection IDs, per connection, would be for hardware offload.

Reply:
I tend to agree. If it were the case that a connection ID was path specific it would be a different matter, but it is required to work for all paths. It could be argued that removing the ability to change length would complicate multi-path design later on - but that is hard to predict.

Another point if view is that a connection ID might be related to infrastructure provided and it might change suddenly for a large number of live connections. But still, it seems a bit contrived.